crates/matrix-sdk-common/src/linked_chunk/mod.rs

```diff
@@ -208,9 +208,13 @@ impl<const CAP: usize, Item, Gap> Ends<CAP, Item, Gap> {
         }
     }
 
-    /// Drop all chunks, and replace the first one with the one provided as an
-    /// argument.
-    fn replace_with(&mut self, first_chunk: NonNull<Chunk<CAP, Item, Gap>>) {
+    /// Clear all the chunks, leaving the chunk in an uninitialized state,
+    /// because `Self::first` is a dangling pointer.
```

Comment:
To help me understand, I would add:

```
/// After this method has run, all items in the `LinkedChunk` have been dropped.
```

(Assuming I'm correct.)

Reply:
(Or maybe just say "Drop all the chunks" in the first comment instead of "Clear all the chunks"?)

Reply:
Thanks! opened https://github.com/matrix-org/matrix-rust-sdk/pull/4845 with a wording change.